### PR TITLE
feat: Implemented ReCaptcha Verification in sign_up process

### LINKED ISSUE #2350: Add ReCaptcha for Signup attempt
Describe the feature you'd like
Add ReCaptcha confirmation for the signup process.

Additional context
fossasia/susi_server@0d13717

Reference master issue: #2319

Would you like to work on the issue?
Yes

### PR DESCRIPTION
**Pull request by s-ayush2903**
Fixes #2350 

Changes: Some dependencies required for ReCaptcha implementation were added, and corresponding changes were made accordingly.

I am with this embedding two screenRecordings that display the proposed changes.

  (This represents the state of the app when the key is present)
GIF-200109_151419

   (And This represents the state of the app when the key is not present)
GIF-200109_152147

-

### CODE REVIEW

**Comment by atm1504** [comment]
Why did you create a pr for an issue that has already been claimed by someone else, who has already made a pr?

**Comment by s-ayush2903** [comment]
> Why did you create a pr for an issue that has already been claimed by someone else, who has already made a pr?

My apologies, Sorry!

**Comment by s-ayush2903** [comment]
As the above pull request makes the changes that were asked to make in the #2360 and to satisfy which, another #2444 was opened and neither of which includes the requested changes like ReCaptcha_Key storage issue and the build issue(e.g. the build fails in both of the previous MRs when key is not supplied) and the issue related to usage of timber, so REOPENING it.

**Comment by s-ayush2903** [comment]
@liveHarshit please review

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 1]
```diff
@@ -250,23 +255,37 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
...
+                            email.error = null
+                            password.error = null
+                            confirmPassword.error = null
+                            inputUrlSignUp.error = null
+
+                            val stringEmail = email.editText?.text.toString()
+                            val stringPassword = password.editText?.text.toString()
+                            val stringConfirmPassword = confirmPassword.editText?.text.toString()
+                            val stringURL = inputUrlSignUp.editText?.text.toString()
+                            Log.d("Great", "Signed up")
```
Comment (atm1504) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Remove

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt [hunk 2]
```diff
@@ -144,6 +144,7 @@ class SignUpPresenter(signUpActivity: SignUpActivity) : ISignUpPresenter, ISignU
         databaseRepository.deleteAllMessages()
         settingView?.startLoginActivity()
     }
+
```
Comment (atm1504) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: revert

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 3]
```diff
@@ -100,8 +105,8 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
                     startActivity(intent)
                     finish()
                 })
-                setNegativeButton(android.R.string.no, DialogInterface.OnClickListener {
-                    dialog, id -> dialog.cancel()
+                setNegativeButton(android.R.string.no, DialogInterface.OnClickListener { dialog, id ->
```
Comment (atm1504) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Revert

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -115,6 +115,7 @@
             android:name=".chat.YouTubeActivity"
             android:theme="@style/Theme.AppCompat.Dialog" />
 
+
```
Comment (atm1504) [inline app/src/main/AndroidManifest.xml]: Remove unwanted line changes

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt (hunk 1 above)
Comment (s-ayush2903) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: updated

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt (hunk 3 above)
Comment (s-ayush2903) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: done

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt (hunk 2 above)
Comment (s-ayush2903) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: updated

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (s-ayush2903) [reply app/src/main/AndroidManifest.xml]: done

Comment (s-ayush2903) [reply app/src/main/AndroidManifest.xml]: dine

**Comment by mkmar01** [comment]
@s-ayush2903 status ?

**Comment by angmas1** [comment]
@s-ayush2903 are you still working on this?

**Pull request closed without merging**